subsys/dsp/Kconfig

```diff
@@ -13,10 +13,16 @@ if DSP
 config DSP_BACKEND_HAS_STATIC
 	bool
 
+config DSP_POINTER_HAS_RESTRICT
+	bool "Use restrict type qualifier"
+	help
+	  Restrict keyword indicates that a pointer isn't aliased in the current scope.
+
 choice DSP_BACKEND
 	prompt "DSP library backend selection"
 	default DSP_BACKEND_CMSIS if CMSIS_DSP
 	default DSP_BACKEND_CUSTOM
+	default DSP_BACKEND_ARC
```

Comment:
Since there is no `if` above this, this default will never trigger. I think you should move the default up so it looks like:

```
default DSP_BACKEND_ARC if ARC
default DSP_BACKEND_CUSTOM
```

Reply:
I guess it's even need to be

```
default DSP_BACKEND_ARC if ARC && "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "arcmwdt"
default DSP_BACKEND_CUSTOM
```

as this is dsp backend from arc mwdt toolchain.